cpu/stm32/periph/spi_u5.c

```diff
@@ -0,0 +1,386 @@
+  /* reset configuration */
+  dev(bus)->CR1 = 0;
+  dev(bus)->CFG1 = SPI_CFG1_SETTINGS;
+  dev(bus)->CR2 = SPI_CR2_SETTINGS;
+  dev(bus)->CFG2 = SPI_CFG2_MASTER;
+  periph_clk_dis(spi_config[bus].apbbus, spi_config[bus].rccmask);
+}
+
+void spi_init_pins(spi_t bus) {
+  assume(bus < SPI_NUMOF);
+  if (gpio_is_valid(spi_config[bus].sclk_pin)) {
+    gpio_init_af(spi_config[bus].sclk_pin, spi_config[bus].sclk_af);
+  }
+
+  if (gpio_is_valid(spi_config[bus].mosi_pin)) {
+    gpio_init_af(spi_config[bus].mosi_pin, spi_config[bus].mosi_af);
+  }
+
+  if (gpio_is_valid(spi_config[bus].miso_pin)) {
+    gpio_init(spi_config[bus].miso_pin, GPIO_IN);
+    gpio_init_af(spi_config[bus].miso_pin, spi_config[bus].miso_af);
+  }
+
+  if (gpio_is_valid(spi_config[bus].cs_pin)) {
+    gpio_init_af(spi_config[bus].cs_pin, spi_config[bus].cs_af);
+  }
+  mutex_unlock(&locks[bus]);
+}
+
+void spi_deinit_pins(spi_t bus) {
+  assume(bus < SPI_NUMOF);
+  mutex_lock(&locks[bus]);
+
+  if (gpio_is_valid(spi_config[bus].sclk_pin)) {
+    gpio_init(spi_config[bus].sclk_pin, GPIO_IN);
+  }
+
+  if (gpio_is_valid(spi_config[bus].mosi_pin)) {
+    gpio_init(spi_config[bus].mosi_pin, GPIO_IN);
+  }
+
+  if (gpio_is_valid(spi_config[bus].miso_pin)) {
+    gpio_init(spi_config[bus].miso_pin, GPIO_IN);
+  }
+}
+
+gpio_t spi_pin_miso(spi_t bus) {
+  assume(bus < SPI_NUMOF);
+  return spi_config[bus].miso_pin;
+}
+
+gpio_t spi_pin_mosi(spi_t bus) {
+  assume(bus < SPI_NUMOF);
+  return spi_config[bus].mosi_pin;
+}
+
+gpio_t spi_pin_clk(spi_t bus) {
+  assume(bus < SPI_NUMOF);
+  return spi_config[bus].sclk_pin;
+}
+
+int spi_init_cs(spi_t bus, spi_cs_t cs) {
+  if (bus >= SPI_NUMOF) {
+    return SPI_NODEV;
+  }
+  if (!gpio_is_valid(cs) ||
+      (((cs & SPI_HWCS_MASK) == SPI_HWCS_MASK) && (cs & ~(SPI_HWCS_MASK)))) {
+    return SPI_NOCS;
+  }
+
+  if (cs == SPI_HWCS_MASK) {
+    if (!gpio_is_valid(spi_config[bus].cs_pin)) {
+      return SPI_NOCS;
+    }
+    gpio_init(spi_config[bus].cs_pin, GPIO_OUT);
+    gpio_init_af(spi_config[bus].cs_pin, spi_config[bus].cs_af);
+  } else {
+    gpio_init((gpio_t)cs, GPIO_OUT);
+    gpio_set((gpio_t)cs);
+  }
+
+  return SPI_OK;
+}
+
+void spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk) {
+  assume((unsigned)bus < SPI_NUMOF);
+
+  /* lock bus */
+  mutex_lock(&locks[bus]);
+#ifdef STM32_PM_STOP
+  /* block STOP mode */
+  pm_block(STM32_PM_STOP);
+#endif
+  /* enable SPI device clock */
+  periph_clk_en(spi_config[bus].apbbus, spi_config[bus].rccmask);
+  /* enable device */
+  if (clk != clocks[bus]) {
+    prescalers[bus] = _get_prescaler(&spi_config[bus], clk);
+    clocks[bus] = clk;
+  }
+  uint8_t br = prescalers[bus];
+
+  DEBUG("[spi] acquire: requested clock: %" PRIu32
+        " Hz, resulting clock: %" PRIu32 " Hz, BR prescaler: %u\n",
+        clk, periph_apb_clk(spi_config[bus].apbbus) >> (br + 1), (unsigned)br);
+  uint32_t cfg1 = br << BR_SHIFT;
```

Comment:
This clobbers the dsize reg and sets the SPI back to 4 bit mode, among other things. Maybe change to:
```suggestion
  uint32_t cfg1 = SPI_CFG1_SETTINGS | (br << BR_SHIFT);
```